newrelic/resource_newrelic_alert_policy.go

```diff
@@ -78,6 +108,12 @@ func resourceNewRelicAlertPolicyRead(d *schema.ResourceData, meta interface{}) e
 		return err
 	}
 
+	channelIDs := d.Get("channel_ids").([]interface{})
+
+	if len(channelIDs) > 0 {
+		d.Set("channel_ids", expandChannelIDs(channelIDs))
```

Comment:
Fairly certain we don't need to  this (lines 111-115) because state is still preserved due to the fact that if a user adds or removes any channel IDs, the original resource is destroyed and then recreated with the updated channel associations (via the `ForceNew: true` schema option). @ctrombley @zlesnr any thoughts on that?

Reply:
Correct, however since any changes to channel IDs warrants a new resource, this is never needed because the state will be saved on create. I guess it doesn't hurt to keep this here. I know that if I remove this part, it doesn't affect anything, but again, it's probably fine to leave it, especially since it's a bit more explicit.

Reply:
I think you're right - this conditional only evals true when we already are tracking ids within a policy, and since we are forcing new on any change, it's not needed.  👍

Reply:
Awesome, thanks for the affirmation